### PR TITLE
[SYCL] fix post-commit failure introduced by override of get_info<parent_device>

### PR DESCRIPTION
**Pull request by cperkinsintel**
gcc and clang++ differ in warnings for duplicate specializations. get_info<parent_device> is overridden, leading to a duplicate specialization, leading to failures in post-commit where clang++ is used to compile.  

This PR removes the duplication entirely. 

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

**Comment by cperkinsintel** [comment]
Post Commit check running here: https://github.com/intel/llvm/actions/runs/3396476032

File: sycl/include/sycl/detail/info_desc_helpers.hpp [hunk 1]
```diff
@@ -95,6 +95,9 @@ struct IsSubGroupInfo<info::kernel_device_specific::compile_sub_group_size>
     using return_type = info::DescType::Desc::return_type;                     \
   };
 #include <sycl/info/device_traits.def>
+// requires explicit handling
+__SYCL_PARAM_TRAITS_SPEC(device, parent_device, sycl::device,
+                         PI_DEVICE_INFO_PARENT_DEVICE)
```
Comment (aelovikov-intel) [inline sycl/include/sycl/detail/info_desc_helpers.hpp:100]: A better approach/pattern to such special handling is, IMO:

```
// .def:
#ifndef __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...)
#define __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...) __SYCL_PARAM_TRAITS_SPEC(...)
#endif

// use_normal.cpp
#define __SYCL_PARAM_TRAITS_SPEC(....) ...
#include 

// use_special.cpp
#define __SYCL_PARAM_TRAITS_SPEC(....) ....
#define __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...)
#include
```

That said, I'm not familiar with the code to comment on whether it does the right thing. I can only review how it does what it's doing.

Comment (cperkinsintel) [reply sycl/include/sycl/detail/info_desc_helpers.hpp:100]: That's a good suggestion. I was hurrying just trying to fix the post-commit failure. Let me check it out. If it's easy enough, I'll do it now. If not, I'll open a JIRA and do it next week.

Comment (cperkinsintel) [reply sycl/include/sycl/detail/info_desc_helpers.hpp]: done

**Comment by cperkinsintel** [comment]
Post Commit check (after incorporating feedback from @aelovikov-intel  is running here: https://github.com/intel/llvm/actions/runs/3397955702

**Comment by pvchupin** [comment]
@cperkinsintel there are few issues left with ABI symbols, can you please check if these are related?

**Pull request merged**